testinfra/backend/winrm.py

```diff
@@ -0,0 +1,66 @@
+# coding: utf-8
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import unicode_literals
+from __future__ import absolute_import
+
+from testinfra.backend import base
+
+try:
+    import winrm
+except ImportError:
+    raise RuntimeError((
+        "You must install the pywinrm package (pip install pywinrm) "
+        "to use the winrm backend"))
+
+import winrm.protocol
+
+
+class WinRMBackend(base.BaseBackend):
+    """Run command through winrm command"""
+    NAME = "winrm"
+
+    def __init__(self, hostspec, password=None, ssl=True, verify_ssl=True,
+                 transport=None, *args, **kwargs):
+        self.host, self.username, self.port = self.parse_hostspec(hostspec)
+        self.password = password
+        self.ssl = ssl
+        self.verify_ssl = verify_ssl
+        self.transport = transport or 'ntlm'
```

Comment:
I'd use ``__init__(..., transport='ntlm') and self.transport = transport`` here.

Reply:
ahh yeah, strings are immutable, so that makes sense, just habit with passing dictionaries around probably